enaml/core/declarative_meta.py

```diff
@@ -5,7 +5,10 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 #------------------------------------------------------------------------------
-from atom.api import Atom, AtomMeta, DefaultValue, Member, Typed
+from atom.api import Atom, AtomMeta, ChangeType, DefaultValue, Member, Typed
+
+# The declarative engine only updates for these types of changes
+D_CHANGE_TYPES = ChangeType.UPDATE | ChangeType.PROPERTY | ChangeType.EVENT
```

Comment:
I feel like the previous version was more sensible, but let see if @sccolbert has an opinion.

Reply:
After thinking about it a little, the only way it could send the wrong change types would be if someone re-adds the `declarative_change_handler` with different flags.

Since that is unlikley to happen unintentially, I figured its safe to do.

I added a new set of flags for the observe decorator so that still keeps everything except create.

Reply:
I wanted to write and test a minimal demonstration, but since I can not find the time, let me try to demonstrate why I believe delete and container should be on that list.

```python
enamldef A(Container): a:
    attr t: str = "test"
    
    Field:
        test := a.t

    PushButton:
        text = "Reset"
        clicked:: del a.t
```

I often use the ability of atom to delete an attribute to restore it to its default. If the declarative engine ignores delete, it won't update the field here, which I believe is wrong.

WDYT ? I think a similar argument holds for container.

Reply:
They weren't used before unless I'm missing something: 
https://github.com/nucleic/enaml/blob/main/enaml/core/declarative_meta.py#L644

I think it'd be best to keep it the same then try enabling those in a separate PR.

Reply:
We could also check both project history. It is possible this bit predates delete and container notifications.

Reply:
It looks like delete was there 6 months before the property one was added. 

I guess I can just try it and see if there's any noticeable difference.

Reply:
Please do. Short of a direct answer from @sccolbert, this is the best path forward since we both agree it would make sense for it to be present.

Reply:
I don't think I ever enabled container event listening, because I was never really happy with my implementation of reactive containers. For example, someone calling `.append` in a loop on a container list would end in disaster if the events were fired off synchronously. 

If I remember correctly, I was intending to add a dispatch mechanism which would batch the changes into a single event, then fire off that event when control returned to the event loop. But that would be a pretty big can of worms, probably best left to a major update/revision where we can streamline a bunch of stuff by not trying to be toolkit backend agnostic.

Reply:
I understand this concern. So I propose that for this PR we enable delete, leave container out and add somewhere in tye documentation the special behavior of enaml with respect to create and container.

Reworking notification should first happen in atom and while this is something I have been thinking about for years, I do not seem to be able to find the time to make any progress on it.